lib/fs.js

```diff
@@ -1358,6 +1358,9 @@ fs.appendFile = function(path, data, options, callback) {
     throwOptionsError(options);
   }
 
+  // make a copy of options before making changes
```

Comment:
Would it make sense to enforce this if https://github.com/eslint/eslint/issues/6055 land?